windows/src/engine/keyman32-ver/calldll.cpp

```diff
@@ -575,11 +587,50 @@ LoadDLLs(LPINTKEYBOARDINFO lpkbi) {
 
   if (!GetKeyboardFileName(lpkbi->Name, fullname, _MAX_PATH))
     return FALSE;
-  if ((!lpkbi->lpCoreKeyboard) || (!lpkbi->lpCoreKeyboard)){
+  if (!lpkbi->lpCoreKeyboard){
+    return FALSE;
+  }
+
+  PKEYMAN64THREADDATA _td = ThreadGlobals();
+  if (!_td){
     return FALSE;
   }
 
   km_kbp_keyboard_imx *imx_list = lpkbi->lpIMXList;
+  // return early if the list empty avoiding loading the proxy dll into memory
+  if (!imx_list->library_name) {
+    return FALSE;
+  }
+
+  char versioned_filename[_MAX_PATH];
+  char proxy_modulename[_MAX_PATH];
+  char check_modulename[_MAX_PATH];
+
+  GetModuleFileName(g_hInstance, versioned_filename, _MAX_PATH);
+
+  char *p = strrchr(versioned_filename, '\\'); // find last occurrence of '\'
+  if (p != NULL) {
+      int64_t len = p - versioned_filename + 1;
+      strncpy(proxy_modulename, versioned_filename, (size_t)len);
+      strncat(proxy_modulename, keyman_engine_arch, 13);
+  }
+  if (_td->hModuleProxy){ // Not sure I need this check
+    GetModuleFileName(_td->hModuleProxy, check_modulename, _MAX_PATH);
+    if (!strstr(check_modulename, proxy_modulename))
+    {
+      FreeLibrary(_td->hModuleProxy);
+      _td->hModuleProxy = LoadLibrary(proxy_modulename);
+    }
+  }
+  else {
+    _td->hModuleProxy = LoadLibrary(proxy_modulename);
+  }
```

Comment:
The filename of the proxy DLL shouldn't change, so we shouldn't need to unload and reload?

```suggestion
  if (!_td->hModuleProxy) {
    char versioned_filename[_MAX_PATH];
    GetModuleFileName(g_hInstance, module_filename, _MAX_PATH); // TODO: check result
  
    char *p = strrchr(module_filename, '\\'); // find last occurrence of '\'
    if (p != NULL) {
      int64_t len = p - module_filename;
      // We know that the arch filename is shorter than the versioned filename so this
      // should always be safe
      strncpy(p, keyman_engine_arch, _MAX_PATH - len - 1); // TODO: -1? need to check
      _td->hModuleProxy = LoadLibrary(module_filename);
    }
  }
```